model/scratch/tests/test_ridge.py

```diff
@@ -0,0 +1,93 @@
+]
+
+class TestRidge(AsyncTestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.model_dir = tempfile.TemporaryDirectory()
+        cls.feature1 = DefFeature("Years", float, 1)
+        cls.feature2 = DefFeature("Expertise", float, 1)
+        cls.feature3 = DefFeature("Trust", float, 1)
+        cls.features = ['Expertise', 'Trust', 'Years']
+        cls.repos = [
+            Repo(str(i), data={"features": Features_data[i]})
+            for i in range(0, len(Features_data))
+        ]
+        cls.sources = Sources(
+            MemorySource(MemorySourceConfig(repos=cls.repos))
+        )
+        cls.model = RidgeRegression(
+            RidgeConfig(
+                directory=cls.model_dir.name,
+                predict=DefFeature("Salary", float, 1),
+                features=cls.features,
+            )
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.model_dir.cleanup()
+
+    async def test_context(self):
+        async with self.sources as sources, self.model as model:
+            async with sources() as sctx, model() as mctx:
+                # Test train
+                await mctx.train(sctx)
+                # Test accuracy
+                res = await mctx.accuracy(sctx)
+                self.assertTrue(0.0 <= res < 1.0)
+                # Test predict
+                target_name = model.config.predict.NAME
+                async for repo in mctx.predict(sctx.repos()):
+                    correct = Features_data[int(repo.key)]["Salary"]
+                    # Comparison of correct to prediction to make sure prediction is within a reasonable range
+                    prediction = repo.prediction(target_name).value
+                    self.assertGreater(prediction, correct - (correct * 0.20))
+                    self.assertLess(prediction, correct + (correct * 0.20))
+
+    async def test_00_train(self):
+        async with self.sources as sources, self.model as model:
+            async with sources() as sctx, model() as mctx:
+                await mctx.train(sctx)
+
+    async def test_01_accuracy(self):
+        async with self.sources as sources, self.model as model:
+            async with sources() as sctx, model() as mctx:
+                res = await mctx.accuracy(sctx)
+                self.assertTrue(0.0 <= res < 1.0)
+
+    async def test_02_predict(self):
+        async with self.sources as sources, self.model as model:
+            async with sources() as sctx, model() as mctx:
+                target_name = model.config.predict.NAME
+                async for repo in mctx.predict(sctx.repos()):
+                    target_name = model.config.predict.NAME
+                    correct = Features_data[int(repo.key)]["Salary"]
+                    prediction = repo.prediction(target_name).value
+                    self.assertGreater(prediction, correct - (correct * 0.20))
+                    self.assertLess(prediction, correct + (correct * 0.20))
```

Comment:
Great! Let's also add a CLI style integration test for this, [example](https://github.com/intel/dffml/blob/f61fa837013cb7b53d6d6190c18bc79f66d4af3c/model/scikit/tests/test_scikit_integration.py#L117-L212)

Reply:
ok